dark/sam.py

```diff
@@ -1,14 +1,17 @@
 import sys
-import six
 import numpy as np
+from typing import Optional
```

Comment:
I'm not sure when `Optional` was introduced. https://docs.python.org/3/library/typing.html only says it was changed in 3.10. Kicking this can down the road.